### PR TITLE
Bulk filter/column add feature

### PR DESCRIPTION
**Pull request by Akrion**
- Adds bulk add support to the NestedPicker
- UI tweaks on the associated components to make them more usable

CleanShot 2021-08-09 at 21 12 57@2x

CleanShot 2021-08-09 at 21 13 39@2x

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 2848cae0f0cb79de58d6558aa057c984efe26929


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6ec4f6cfd3648ed1a85d30dfc279b5f577e87205
</p>

**Review by stellarhoof — CHANGES_REQUESTED** [review]

File: src/greyVest/NestedPicker.js [hunk 1]
```diff
@@ -113,25 +148,73 @@ let NestedPicker = ({
...
+    margin: 0,
+    padding: 0,
+    minWidth: 500,
+    maxHeight: 400,
+    paddingBottom: 10,
+  },
+  theme: { Button },
 }) => {
-  let filter = React.useState('')
+  let state = observable({
```
[the remark this answers is not in the document]
Comment (Akrion) [reply src/greyVest/NestedPicker.js:161]: Did you try it? Use the storybook ...

[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/greyVest/NestedPicker.js:161]: I'm doing that now and it works but that's not the point. This pattern is just very fragile. The component will break on a rerender. That's why there's a hook provided by mobx-react that uses internal state so you always get the same observable store.

[the remark this answers is not in the document]
Comment (Akrion) [reply src/greyVest/NestedPicker.js:161]: Which hook is that? https://github.com/mobxjs/mobx-react#uselocalobservable-hook?

[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/greyVest/NestedPicker.js:161]: `useLocalObservable`: https://mobx.js.org/react-integration.html#using-local-observable-state-in-observer-components

[the remark this answers is not in the document]
Comment (Akrion) [reply src/greyVest/NestedPicker.js:161]:
```
WARNING in ./src/greyVest/NestedPicker.js 176:14-32 "export 'useLocalObservable' was not found in 'mobx-react'
```

[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/greyVest/NestedPicker.js:161]: It's either in mobx-react-lite or we have to bump mobx-react

**Review by stellarhoof — APPROVED** [review]
This is such a nice improvement on the picker :)

File: src/greyVest/NestedPicker.stories.js [hunk 2]
```diff
@@ -3,12 +3,10 @@ import { storiesOf } from '@storybook/react'
...
+import { options } from '../utils/stories'
 
 storiesOf('GreyVest Library|NestedPicker', module)
   .addDecorator(decorator)
   .add('story', () => (
-    <NestedPicker
-      options={['abcd', 'bcde', 'cdef'].map(x => ({ label: x, value: x }))}
-      onChange={action(`picked`)}
-    />
+    <NestedPicker options={options} onChange={x => action(`picked`)(x)} />
```
Comment (ccalbaugh) [inline src/greyVest/NestedPicker.stories.js:11]: Is there a reason the onChange got wrapped with the extra function?  It doesn't seem necessary

**Pull request merged**